### DISCUSSION TITLE
part-ii/2017-27/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#396] [Question comments]
Part II 2017 Automata and Formal Languages: Paper 3, Section II, $11 \mathrm{H}$ Automata and formal languages

https://questions.tripos.org/part-ii/2017-27/

**Comment by artie2000** [comment]
All but (c) is bookwork.

<details>
<summary>(c):</summary>

Idea: map $W_n$ to the first $|W_n|$ elements of $\mathbb{K}$.

Formally: map

$g(m,n)=\left.\begin{cases}e(m)& |W_n|\geq m\\\uparrow& |W_n|<m\end{cases}\right\}$, where $e:\mathbb{N}\to\mathbb{K}$ is a (computable) enumeration of $\mathbb{K}$.

Then s-m-n directly gives the reduction we want.

To construct $g$, diagonalise $\varphi_\mathbb{K}$ and $f_{n,1}$, and return the $m^\text{th}$ element of $\mathbb{K}$ after $m$ inputs of $f_{n,1}$ have terminated.

</details>
